pep-0622.rst

```diff
@@ -1184,50 +1170,32 @@ ambiguous with capture patterns. Four other alternatives were considered:
   to be assigned to.  An earlier version of this proposal used a
   leading-dot rule::
 
-    FOO = 1
-    value = 0
-
-    match value:
-        case .FOO:  # This would not be matched
-            ...
-        case BAR:  # This would be matched
-            ...
+    match entree[-1]:
+        case .spam: ...  # Compares entree[-1] == spam.
+        case side: ...   # Assigns side = entree[-1].
 
   While potentially useful, it introduces strange-looking new syntax
   without making the pattern syntax any more expressive.  Indeed,
   named constants can be made to work with the existing rules by
   converting them to ``Enum`` types, or enclosing them in their own
   namespace (considered by the authors to be one honking great idea)::
 
-    class Consts:
-        FOO = 1
-        ...
-
-    value = 0
-
-    match value:
-        case Consts.FOO:  # This would not be matched
-            ...
-        case BAR:  # This would be matched
-            ...
+    match entree[-1]:
+        case Sides.SPAM: ...  # Compares entree[-1] == Sides.SPAM.
+        case side: ...        # Assigns side = entree[-1].
 
   If needed, the leading-dot rule (or a similar variant) could be
   added back later with no backward-compatibility issues.
 
 * There was also on idea to make lookup semantics the default, and require
-  ``$`` to be used in capture patterns::
-
-    FOO = 1
-    value = 0
+  ``$`` or ``?`` to be used in capture patterns::
 
-    match value:
-        case FOO:  # This would not be matched
-            ...
-        case $BAR:  # This would be matched
-            ...
+    match entree[-1]:
+        case spam: ...   # Compares entree[-1] == spam.
+        case side?: ...  # Assigns side = entree[-1].
 
-  But the capture patterns are more common in typical code, so having special
-  syntax for common case would be weird.
+  But capture patterns are more common in typical code, so it is
+  undesirable to require special syntax for them.
```

Comment:
There are more arguments against marking capture variables with sigils:
- None of the other languages surveyed do this (and there's no precedent in Python either).
- It would look confusing to see
  ```
  case (%x, %y):
      return Point(x, y)  # Why not Point(%x, %y)
  ```